src/test/java/io/jenkins/plugins/ml/IPythonBuilderTest.java

```diff
@@ -68,18 +72,25 @@ public void createMocks() throws Exception {
     @Test
     public void testAdditionBuild() throws Exception {
 
+        WorkflowJob p = jenkins.createProject(WorkflowJob.class, "p");
         ServerJobProperty ijj = new ServerJobProperty("localHost");
         assertNotNull("Job property is null",ijj);
-        project.addProperty(ijj);
-        ServerJobProperty jobProp = project.getProperty(ServerJobProperty.class);
+        p.addProperty(ijj);
+        ServerJobProperty jobProp = p.getProperty(ServerJobProperty.class);
         assertNotNull(jobProp);
-
-        /* Builder test
-        String name = "%text 37";
-        IPythonBuilder builder = new IPythonBuilder("32+5",null,"text");
-        project.getBuildersList().add(builder);
-        FreeStyleBuild build = jenkins.buildAndAssertSuccess(project);
-        jenkins.assertLogContains( name, build);
-        */
+        p.setDefinition(
+                new CpsFlowDefinition(
+                        "node{\n"
+                                + "def testImage = docker.image('loghijiaha/ml-agent')\n"
```

Comment:
@kinow How can we setup python dependencies in CI server by using Jenkins file? I have used our ml-agent image to start the container. The script is running inside the docker. Your error is different than the CI errors. I am a little confused.

Reply:
I think all the dependencies wikm have to be installed in the container. Nothing in the box running the CI build.

Reply:
Or maybe there is something in the Jenkins Wiki about that. I've used Jenkins in-house only. On GH always Travis or GH actions. So not sure how to troubleshoot errors like this.